### PR TITLE
docs(faq): document the AngularJS/jqLite deprecation strategy

### LINKED ISSUE #15282: Document that jqLite deprecations follow jQuery deprecation timeline
Normally, once a feature is deprecated it will stay in the codebase for one minor release and will be removed in the next minor release (in Angular 1, minor releases are breaking release).

As discussed in #15221, we have decided to treat deprecations in jqLite which are directly related to corresponding deprecations in jQuery differently. Namely, the deprecated feature should **not** be removed, until jQuery removes it too (deprecated features tend to stay longer in jQuery than in Angular).

This decision should be documented in our FAQ.

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Docs update.

**What is the current behavior? (You can also link to an open issue here)**
The deprecation strategy is not specified.

**What is the new behavior (if this is a feature change)?**
The deprecation strategy is specified.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format

**Other information**:

Fixes #15282

### CODE REVIEW

File: docs/content/misc/faq.ngdoc [hunk 1]
```diff
@@ -130,6 +130,15 @@ AngularJS 1.3 only supports jQuery 2.1 or above. jQuery 1.7 and newer might work
 but we don't guarantee that.
 
 
+### When will the deprecated features be removed from the library?
```
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: When **are** ~~will the~~ deprecated features ~~be~~ ...

File: docs/content/misc/faq.ngdoc [hunk 2]
```diff
@@ -130,6 +130,15 @@ AngularJS 1.3 only supports jQuery 2.1 or above. jQuery 1.7 and newer might work
...
+Most of the time we remove a deprecated feature in a next minor bump. For example, the `preAssignBindingsEnabled`
```
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: minor **version** bump

File: docs/content/misc/faq.ngdoc [hunk 3]
```diff
@@ -130,6 +130,15 @@ AngularJS 1.3 only supports jQuery 2.1 or above. jQuery 1.7 and newer might work
...
+`$compileProvider` method was defined in AngularJS `1.5.10`, deprecated in `1.6` and will be removed in `1.7`.
```
Comment (Narretz) [inline docs/content/misc/faq.ngdoc]: We'll have to change this when 1.7.0 is released.

**Review by Narretz — CHANGES_REQUESTED** [review]
Just a few minor improvements in the added text.
I noticed that we basically don't talk about deprecation in the rest of the FAQ. Maybe we should add this to the section about the versioning strategy, and add this new deprecation section as a subsection?

**Comment by mgol** [comment]
> Maybe we should add this to the section about the versioning strategy, and add this new deprecation section as a subsection?

Makes sense.

File: docs/content/misc/faq.ngdoc (hunk 3 above)
Comment (mgol) [inline docs/content/misc/faq.ngdoc]: Hmm, maybe it'd be better to put sth else here, preferably something where we adhered to this strategy. :)

**Comment by mgol** [comment]
@Narretz PR updated, PTAL.

I amended the commit as the diff wouldn't tell much anyway when the whole section is moved.

**Pull request merged**